pkg/clusterclientstore/cache/cluster.go

```diff
@@ -58,12 +65,26 @@ func (c *cluster) IsReady() bool {
 	return c.state == StateReady
 }
 
-func (c *cluster) GetClient() (kubernetes.Interface, error) {
+func (c *cluster) GetClient(ua string) (kubernetes.Interface, error) {
```

Comment:
It might be worth leaving a comment what ua means and what it's expected/valid value.